### ISSUE TITLE
Memory leak in _struct module 

### ISSUE THREAD

**Issue by kumaraditya303** [python/cpython#94207]
Reproducer:

```py
import _struct; _struct.calcsize('b')
```
Memory Leak:
```console
./python -X showrefcount main.py
[367 refs, 150 blocks]
```

Commit tested: 27b9894033

**Comment by mdickinson** [comment]
Possible fix in #94239.

**Comment by mdickinson** [comment]
Fixed in #94239 and its backports. Thanks for the report!
